extide/gradle/src/org/netbeans/modules/gradle/execute/GradleDaemonExecutor.java

```diff
@@ -282,6 +276,40 @@ public void statusChanged(org.gradle.tooling.events.ProgressEvent event) {
         }
     }
 
+    private boolean setPlatformAndEnv(BuildLauncher buildLauncher, GradleJavaPlatformProvider platformProvider, String runEnvironment) {
+        String javaHome = null;
+        if (platformProvider != null) {
+            try {
+                buildLauncher.setJavaHome(platformProvider.getJavaHome());
+                javaHome = platformProvider.getJavaHome().getCanonicalPath();
+            } catch (IOException ex) {
+                io.getErr().println(Bundle.NO_PLATFORM(ex.getMessage()));
+                gradleTask.finish(1);
+                return false;
+            }
+        }
+        if (javaHome != null || runEnvironment != null) {
+            Map<String, String> envs = new HashMap<>(System.getenv());
+            if (javaHome != null) {
```

Comment:
I'd maybe put the `javaHome` last, so it overrides values in the environment acquired from `runEnvironment` project property.
@lkishalmi what's your opinion ?

Reply:
I wasn't sure about the preferred order.

Reply:
To be transparent, I've added a print of a warning: `Warning: JAVA_HOME environment variable is replaced with the current Java platform path <javaHome set to platformProvider.getJavaHome().getCanonicalPath()>`.